### PR TITLE
Add test for max depth in exception message

### PR DESCRIPTION
**Pull request by thomaslevesque**
Relates to #34410

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs [hunk 1]
```diff
@@ -57,7 +58,7 @@ public static void WriteCyclicFail(int objectHierarchyDepth, int maxDepth, int e
             // Since the last Parent property is null, the serializer moves onto the Children property.
             string expectedPath = "$" + string.Concat(Enumerable.Repeat(".Parent", expectedPathDepth));
             Assert.Contains(expectedPath, ex.Path);
-            Assert.Contains(maxDepth.ToString(), ex.ToString());
+            Assert.Contains(effectiveMaxDepth.ToString(), ex.ToString());
```
Comment (Jozkee) [inline src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs:61]:
```suggestion
            Assert.Contains(effectiveMaxDepth.ToString(), ex.Message);
```

`ex.ToString()` also contains the call stack (including line numbers), which could give us false positives.

Comment (thomaslevesque) [reply src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs:61]: Yeah, I thought about this, but didn't want to change the existing test too much. I'll do a more specific check.

Comment (thomaslevesque) [reply src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs]: I amended the commit

File: src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs [hunk 2]
```diff
@@ -57,7 +58,8 @@ public static void WriteCyclicFail(int objectHierarchyDepth, int maxDepth, int e
             // Since the last Parent property is null, the serializer moves onto the Children property.
             string expectedPath = "$" + string.Concat(Enumerable.Repeat(".Parent", expectedPathDepth));
             Assert.Contains(expectedPath, ex.Path);
-            Assert.Contains(maxDepth.ToString(), ex.ToString());
+            string expectedMessage = $"the object depth is larger than the maximum allowed depth of {effectiveMaxDepth}";
+            Assert.Contains(expectedMessage, ex.Message);
```
Comment (layomia) [inline src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs:62]:
```suggestion
            Assert.Contains(effectiveMaxDepth.ToString(), ex.Message);
```

We only assert against targeted parts of the exception messages i.e the content wrapped in parentheses in the `SR` formatted string. For example, see the [string for the exception thrown here](https://github.com/dotnet/runtime/blob/5e6b441047c43b39d0306b21c0b78e0d843742a9/src/libraries/System.Text.Json/src/Resources/Strings.resx). This is to align with potential future globalization efforts where the rest of the string will be translated to various languages.

Comment (thomaslevesque) [reply src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs:62]: Ah, I see. I'll change it back.

Comment (thomaslevesque) [reply src/libraries/System.Text.Json/tests/Serialization/CyclicTests.cs]: Done

**Review by layomia — APPROVED** [review]

**Comment by layomia** [comment]
Thanks.

**Pull request merged**